src/servers/ZoneServer2016/commands/commands.ts

```diff
@@ -2155,7 +2155,62 @@ export const commands: Array<Command> = [
       server.groupManager.handleGroupCommand(server, client, args);
     },
   },
+  {
+    name: "deepcover",
+    permissionLevel: PermissionLevels.ADMIN,
+    execute: async (
+      server: ZoneServer2016,
+      client: Client,
+      args: Array<string>
+    ) => {
+      const newCharacterName = args[0];
+
+      // Validate the input to ensure that it is a single word without special characters
+      const isValidInput = /^[a-zA-Z0-9_]+$/.test(newCharacterName);
+      if (!isValidInput) {
+        server.sendChatText(
+          client,
+          "Invalid input. Please enter a single word without special characters."
+        );
+        return;
+      }
+
+      // Update the client's character name
+      client.character.name = newCharacterName;
 
+      // Wait for one second before running vanish command
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // Set the client's isSpectator state
+      client.character.isSpectator = !client.character.isSpectator;
```

Comment:
But if it's already set to true it will be falsy then ? Is it intentionnal ?

Reply:
yes is intentional to force the name change to become active